### DISCUSSION TITLE
ACM Activation Unable to perform signature

### DISCUSSION THREAD

**Discussion by dev0id** [Ylianst/MeshCentral#5297] [Q&A]
I have generated the setup.bin from MeshCentral and used it with a USB drive to provision MBEx.

When trying to run amtconfig I get the error "Unable to perform signature".
amtconfig output:

18:59:42, LMS tunnel start.
18:59:43, Checking Intel AMT state...
18:59:43, Getting ready for ACM activation...
18:59:44, Unable to perform signature.
18:59:44, LMS tunnel closed.

MeshCentral reports AMT as:
Not Activated (Pre), v16.1.25

How can I go about getting the ACM activation to succeed?

**Comment by revellion** [comment]
I'm having a similar issue aswell.

**Reply by dev0id** [reply]
I found a work around.
https://github.com/Ylianst/MeshCentral/commit/43c16d02f9c5538c1dbc9459eaeff5a4a250d8bd
Reverting those changes (lines 88 & 89) fixed it for me.

After those changes I was able to perform ACM activation without any issues.

**Reply by revellion** [reply]
Thanks for the suggestion, gonna give it a try tomorrow :D

**Comment by Steltek** [comment]
> I found a work around. [43c16d0](https://github.com/Ylianst/MeshCentral/commit/43c16d02f9c5538c1dbc9459eaeff5a4a250d8bd) Reverting those changes (lines 88 & 89) fixed it for me.
> 
> After those changes I was able to perform ACM activation without any issues.

Encountered the same issue on a LENOVO system here and this workaround made it work.

**Reply by si458** [reply]
just wondering if you can confirm for me, IF reverting just the first 2 lines of the commit fixes the issue?
or did u also need to revert the last line of that commit too?

**Reply by Steltek** [reply]
Unfortunately I did not test this (I just patched all 3 lines). I have now un-patched my instance (re-created the docker container) but I don't have any new machines to provision at the moment, so I can't test. (Next one will arrive in about a month.)

**Reply by si458** [reply]
i received my first ever intel amt dell today! refurbed v12, been playing around with it, i came across this exact error 
' 18:59:44, Unable to perform signature.'
simple reverted the first 2 lines in the commit, and away it went no problems!
didnt need to revert the last line as we have an `IF/ELSE` around that line now,
so the bug was the `certEntry.hashAlgorithm` so will do PR/MERGE soon to fix that issue!

**Comment by si458** [comment]
reverted the commit back for all!
https://github.com/Ylianst/MeshCentral/commit/a17fd2f26872c83e6a1ec1a2ac350efdf67552d5

**Reply by graikhel-intel** [reply]
Hi @si458, With respect to https://github.com/Ylianst/MeshCentral/commit/a17fd2f26872c83e6a1ec1a2ac350efdf67552d5, I'm curious what was certEntry.hashAlgorithm returning? It should return either sha1 or sha256, may be something with the casing? Hardcoding it will cause an issue in another scenario where a user adds Root Cert hash via MEBx.

**Reply by si458** [reply]
it just kept returning `null` so i have no idea why?
AMT is all new to me!
i did my AMT ACM activation via USB key for testing
if hardcoding this causes an issue and u can replicate it every time for me,
please do share the steps and i can look into it more...
